api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -111,53 +114,73 @@
   public <C /*>>> extends @NonNull Object*/> SpanContext extract(C carrier, Getter<C> getter) {
     checkNotNull(carrier, "carrier");
```

Comment:
What about these? Especially a `null` carrier might make sense if I use a lambda or anonymous inner class as getter that already holds references to the actual carrier.

Reply:
I think ```extract(getter.getCarrier(), getter)``` is more clean than using `null` here and there

Reply:
I was thinking about:
```java
HttpResponse resp = ...;
SpanContext sctx = format.extract(null, (c, key) -> resp.get(key))
```
vs
```java
HttpResponse resp = ...;
SpanContext sctx = format.extract(resp, (c, key) -> c.get(key))
```

I think both are equally valid, and in any case like mentioned in #895, the API/SDK must not use exceptions for error reporting.

Reply:
I agree this is a valid way to call things. But, let's do that as a separate change; please open an issue about it if you feel we need to change this API to support it.